modules/tinymce/src/core/test/ts/browser/init/ShadowDomEditorTest.ts

```diff
@@ -9,21 +9,17 @@ import { Arr, Strings } from '@ephox/katamari';
 
 const isSkin = (ss: StyleSheet) => ss.href !== null && Strings.contains(ss.href, 'skin.min.css');
 
-UnitTest.asynctest('Skin stylesheets should be loaded in ShadowRoot when editor is in ShadowRoot', (success, failure) => {
+const shadowRootFromEditor = (editor: Editor) => ShadowDom.getShadowRoot(Element.fromDom(editor.getElement())).getOrDie();
 
+UnitTest.asynctest('Skin stylesheets should be loaded in ShadowRoot when editor is in ShadowRoot', (success, failure) => {
   if (!ShadowDom.isSupported()) {
     return success();
   }
 
   Theme();
 
-  const shadowHost = Element.fromTag('div', document);
-  Insert.append(Body.body(), shadowHost);
-  const sr = Element.fromDom(shadowHost.dom().attachShadow({ mode: 'open' }));
-  const editorDiv = Element.fromTag('div', document);
-  Insert.append(sr, editorDiv);
-
-  TinyLoader.setupFromElement((editor: Editor, onSuccess, onFailure) => {
+  TinyLoader.setupInShadowRoot((editor: Editor, onSuccess, onFailure) => {
```

Comment:
Should we maybe pass through the created shadow root in the run callback? It'd save having to look it up all the time.

Reply:
Fixed.